app/src/main/java/org/dhis2/data/forms/dataentry/SearchTEIRepositoryImpl.kt

```diff
@@ -25,20 +29,38 @@ class SearchTEIRepositoryImpl(
         val orgUnitScope = attribute.orgUnitScope() ?: false
 
         if (isUnique && !orgUnitScope) {
-            val teiList = d2.trackedEntityModule().trackedEntityInstanceQuery().onlineOnly()
-                .allowOnlineCache()
-                .eq(true)
-                .byOrgUnitMode()
-                .eq(OrganisationUnitMode.ACCESSIBLE)
-                .byProgram()
-                .eq(programUid)
-                .byAttribute(attribute.uid()).eq(value).blockingGet()
+            try {
+                val teiList = d2.trackedEntityModule().trackedEntityInstanceQuery().onlineOnly()
+                    .allowOnlineCache()
+                    .eq(true)
+                    .byOrgUnitMode()
+                    .eq(OrganisationUnitMode.ACCESSIBLE)
+                    .byProgram()
+                    .eq(programUid)
+                    .byAttribute(attribute.uid()).eq(value).blockingGet()
 
-            if (teiList.isNullOrEmpty()) {
+                if (teiList.isNullOrEmpty()) {
+                    return true
+                }
+
+                return teiList.none { it.uid() != teiUid }
+            } catch (e: Exception) {
+                val exception = if (e.cause != null && e.cause is D2Error) {
```

Comment:
Good job!, but review here the sonar cloud analysis, you have a code smell related with the size